### DISCUSSION TITLE
Reading machine-generated text from image?

### DISCUSSION THREAD

**Discussion by EllenWasbo** [jsvine/pdfplumber#908] [Q&A] [answered]
Hi,

In https://pypi.org/project/pdfplumber/ it is stated:
"Works best on machine-generated, rather than scanned, PDFs..."
To me that sounds like it could also, if the circumstances are right, read scanned PDFs. And this gave me a hope that pdfplumber could work as some kind of OCR (optical character recognition).
I have machine-generated images with text in it (DICOM files as save screens) and I am exploring the possibility of reading text from selected parts of those images without the use of tesseract or EasyOCR or similar where installing other software in addition is mandatory (the end users of my program have strict IT departments and reading text from images is just a small feature that I'd like to offer). 

So could pdfplumber recognize text if not coded as text at all?

I've tested different approaches, but not succeeded so far, but is there a hope if I could tweek the circumstances?

**Comment by samkit-jain** [comment, marked as answer]
Hi @EllenWasbo Appreciate your interest in the library. By machine-generated, we mean any PDF from which you can copy-paste text. To answer your question, no, pdfplumber cannot recognize the text if it is not embedded into the PDF at all.

**Reply by EllenWasbo** [reply]
Thank you for a quick response. The answer was what I expected, but I had this little hope left :) Thank you for a good tool when used for what it is meant for.
